.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,35 @@
+<!--
+Thanks for contributing to Zulip! Please follow the following pull request
+template to help maintainers review your code and merge your changes quicker.
+
+To learn more about this pull request template, read the pull request template
+guide at https://zulip.readthedocs.io/en/latest/pull-request-template.html.
+-->
```

Comment:
This whole file (35 lines in this version) gets copied into the PR message box when someone starts opening a PR, right?

I share @timabbott 's feeling that a long template like this is likely to be off-putting. For a first-time contributor, it looks like this big wall of stuff that I have to read and understand -- probably go follow the link and read that doc -- which is going to look like a lot of work, and has a high chance of causing me not to finish sending the PR. Generally we'd much rather have that first-time PR even if it's not quite right in all respects; after that point a conversation has started, we're able to talk to the person, and they've put a bit of investment in so it's easier to get them to take the time to finish. And for a lot of small polish issues we may just fix them ourselves before merging, so the contributor can pick those details up later.

And for a repeat contributor, it's just a big thing that looms up at me every time.

Here's a sketch of a format I think would work well:
```
<!-- Thanks for your contribution!
Some quick reminders below. For more details, see
https://zulip.readthedocs.io/en/latest/pull-request.html

* Review the PR yourself first and fix typos, messy whitespace, etc.
* Follow Zulip's style for commit messages and independent logical commits.
* Include screenshots in the PR for any UI changes.
* Include tests, and include doc updates for any user-facing feature changes.
-->
```